Anature/src/application/trainers/TrainerBuilder.java

```diff
@@ -221,8 +221,8 @@ private boolean buildIsComplete()
 	private static IAI parseAi(String aiHealthThreshold, String aiSwitchThreshold, String aiMoveThreshold)
 	{
 		double healthThreshold = Double.parseDouble(aiHealthThreshold);
-		AttackEffectiveness switchThreshold = AttackEffectiveness.valueOf(aiSwitchThreshold);
-		AttackEffectiveness moveThreshold = AttackEffectiveness.valueOf(aiMoveThreshold);
+		TypeEffectiveness switchThreshold = TypeEffectiveness.valueOf(aiSwitchThreshold);
+		TypeEffectiveness moveThreshold = TypeEffectiveness.valueOf(aiMoveThreshold);
```

Comment:
Do to the fact that the enums for type effectiveness don't have spaces in them, the game is unable to create trainers with "super effective" of "not effective" thresholds. So you either have to add some code here to remove spaces in the threshold strings (convert "Super Effective" into "SuperEffective") or change the entries in the database to not have spaces.

Reply:
I added spaces to the enums so that can parse correctly

Reply:
You still need to change the database entries to reflect the change. As of now the database is still searching for "Super Effective" instead of "Super_Effective". You could also try using a custom valueOf method (would have to be named something else since you can't override that method) and replace spaces with "_". Either or will work.
If you want to check if the fix works, go into the wild grass. The Wild trainer is designed for the "Not Effective" enum.